File: cpp/263.UglyNumber.cpp
class Solution {
public:
    bool isUgly(int num) {
        if (num<1) return false;
        int prime[4] = {2,3,5,7};
        int ppos = 0;
        while(num!=1)
        {
            if (num%prime[ppos]) ppos++;
            else num/=prime[ppos];
            if (ppos==3) break;
        }
        if (ppos==3) return false;
        return true;
    }
};

File: python/263.UglyNumber.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        
        if num<1:
            return False  #:(
        prime = [2,3,5,7]
        ppos = 0
        while num != 1:
            if num%prime[ppos]:
                ppos += 1
            else:
                num /= prime[ppos]
            if prime[ppos] == 7:
                break
        if prime[ppos]==7:
            return False
        return True
